build-tests/heft-json-schema-typings-plugin-test/src/test/JsonSchemaTypingsGenerator.test.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { FileSystem, type FolderItem, PackageJsonLookup } from '@rushstack/node-core-library';
+
+async function* getFolderItemsAsync(
+  absolutePath: string,
+  relativePath: string
+): AsyncGenerator<[string, string]> {
+  const folderItems: FolderItem[] = await FileSystem.readFolderItemsAsync(absolutePath);
+  for (const item of folderItems) {
+    const itemName: string = item.name;
+    const itemAbsolutePath: string = `${absolutePath}/${itemName}`;
+    const itemRelativePath: string = `${relativePath}/${itemName}`;
+    if (item.isDirectory()) {
+      yield* getFolderItemsAsync(itemAbsolutePath, itemRelativePath);
+    } else {
+      const itemContents: string = await FileSystem.readFileAsync(itemAbsolutePath);
+      yield [itemRelativePath, itemContents];
+    }
+  }
+}
```

Comment:
Since you aren't operating on them as they are emitted, it's better to just return the array.
```suggestion
async function getFolderItemsAsync(
  absolutePath: string,
  relativePath: string
): Promise<Record<string, string>[]> {
  const folderQueue: [string, string][] = [[absolutePath, relativePath]];
  const results: [string, string][] = [];
  for (const [folderAbsolutePath, folderRelativePath] of folderQueue) {
    const folderItems: FolderItem[] = await FileSystem.readFolderItemsAsync(folderAbsolutePath);
    for (const item of folderItems) {
      const itemName: string = item.name;
      const itemAbsolutePath: string = `${folderAbsolutePath}/${itemName}`;
      const itemRelativePath: string = `${folderRelativePath}/${itemName}`;
      if (item.isDirectory()) {
        folderQueue.push([itemAbsolutePath, itemRelativePath]);
      } else {
        const itemContents: string = await FileSystem.readFileAsync(itemAbsolutePath);
        results.push([itemRelativePath, itemContents]);
      }
    }
  }
  results.sort(([a], [b]) => (a < b ? -1 : a > b ? 1 : 0));
  return Object.fromEntries(results);
}
```